homeassistant/helpers/entity.py

```diff
@@ -143,15 +145,30 @@ def update_ha_state(self, force_refresh=False):
 
         If force_refresh == True will update entity before setting state.
         """
+        # We're already in a thread, do the force refresh here.
+        if force_refresh:
+            self.update()
+
+        run_coroutine_threadsafe(
+            self.async_update_ha_state(), self.hass.loop).result()
+
+    @asyncio.coroutine
+    def async_update_ha_state(self, force_refresh=False):
+        """Update Home Assistant with current state of entity.
+
+        If force_refresh == True will update entity before setting state.
+
+        This method must be run in the event loop.
+        """
         if self.hass is None:
             raise RuntimeError("Attribute hass is None for {}".format(self))
 
         if self.entity_id is None:
             raise NoEntitySpecifiedError(
                 "No entity id specified for entity {}".format(self.name))
 
-        if force_refresh:
-            self.update()
+        # if force_refresh:
+        #     yield from self.hass.loop.run_in_executor(None, self.update)
```

Comment:
Rather have this in _our_ worker pool vs being in the executor but how do we communicate to the event loop that a job is done from a job?